packages/admin/src/components/bindingFacade/richText/blockEditor/editor/overrideInsertData.ts

```diff
@@ -97,9 +57,34 @@ export const overrideInsertData = <E extends BlockSlateEditor>(editor: E, option
 			const reference = getEmbedReference()
 			embedHandler!.datum.populateEmbedData({
 				embedArtifacts,
-				source: text,
+				source,
 				entity: reference,
 			})
 		})
 	}
+
+	editor.insertData = data => {
+		if (data.getData('application/x-slate-fragment')) {
+			return insertData(data)
+		}
+
+		const text = data.getData('text/plain').trim()
+		if (!text) {
+			return insertData(data)
+		}
+
+		const url = parseUrl(text)
+
+		for (const [, handler] of options.embedHandlers!) {
+			const result = handler.datum.handleSource(text, url)
+			// because Slate does not expect insertData function to be async
+			// it is probably safer to process embed synchronously if possible
+			if (result && typeof result.then === 'function') {
+				return (result as Promise<any>).then(it => insertEmbed(handler, it, text))
```

Comment:
If I understand correctly, the first embed handler which returns a promise will be always threated as if it can handle the data. 

So this won't work if there is more `embedHandlers` or the promise returned by `handleSource` resolves to `undefined`.